### PR TITLE
Add a quit button to connection lost panel

### LINKED ISSUE #12314: Can't get to menu after loosing connection
When connection to a server is lost and the "Connection Lost..." message box is shown, it seems impossible to exit the game without crashing it (Alt+F4) sometimes (always?).

I first noticed it on the 20161015 and I'm fairly sure it was possible to get to the main menu by pressing Esc earlier.

Experienced on Linux and Windows. 
Sorry if I'm just confused and it's not new.

### PR DESCRIPTION
**Pull request by PunkPun**
Closes #12314

### CODE REVIEW

File: mods/cnc/chrome/connection.yaml [hunk 1]
```diff
@@ -152,7 +152,7 @@ Container@CONNECTION_SWITCHMOD_PANEL:
 			Y: 119
 			Width: 140
 			Height: 35
-			Text: Abort
+			Text: Cancel
```
Comment (Mailaender) [inline mods/cnc/chrome/connection.yaml]: ABORT_BUTTON Text: Cancel?

File: OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs [hunk 2]
```diff
@@ -102,14 +102,16 @@ public class ConnectionFailedLogic : ChromeLogic
...
+		public ConnectionFailedLogic(Widget widget, ModData modData, OrderManager orderManager, NetworkConnection connection, string password, Action onAbort, string abortTextOverwrite, Action<string> onRetry)
 		{
 			var panel = widget;
 			var abortButton = panel.Get<ButtonWidget>("ABORT_BUTTON");
 			var retryButton = panel.Get<ButtonWidget>("RETRY_BUTTON");
 
 			abortButton.Visible = onAbort != null;
 			abortButton.OnClick = () => { Ui.CloseWindow(); onAbort(); };
+			if (abortTextOverwrite != null)
+				abortButton.Text = abortTextOverwrite;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs]: Feels like a hack. Is it really necessary?

File: OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs [hunk 3]
```diff
@@ -102,14 +102,16 @@ public class ConnectionFailedLogic : ChromeLogic
 		bool passwordOffsetAdjusted;
 
 		[ObjectCreator.UseCtor]
-		public ConnectionFailedLogic(Widget widget, ModData modData, OrderManager orderManager, NetworkConnection connection, string password, Action onAbort, Action<string> onRetry)
+		public ConnectionFailedLogic(Widget widget, ModData modData, OrderManager orderManager, NetworkConnection connection, string password, Action onAbort, string abortTextOverwrite, Action<string> onRetry)
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs]: Maybe create an `Action onQuit` and if it is not null show a `QUIT_BUTTON` button?

File: mods/cnc/chrome/connection.yaml (hunk 1 above)
Comment (PunkPun) [reply mods/cnc/chrome/connection.yaml]: it's called cancel in common yaml, and cancel makes more sense here

File: OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs (hunk 3 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs]: definitely an option

File: mods/cnc/chrome/connection.yaml (hunk 1 above)
Comment (Mailaender) [reply mods/cnc/chrome/connection.yaml]: I mean it is inconsistent towards the ID.

Comment (PunkPun) [reply mods/cnc/chrome/connection.yaml]: renamed the one in common to abort

Comment (PunkPun) [reply mods/cnc/chrome/connection.yaml]: Screenshot 2023-03-17 at 09 45 02

Thought maybe `Quit` would be better

**Comment by Mailaender** [comment]
Click Multiplayer - Direct Connect - Join Localhost (without a server running) and crash:

```
Exception of type `System.InvalidOperationException`: ObjectCreator: key `onQuit' not found
   at OpenRA.ObjectCreator.CreateUsingArgs(ConstructorInfo ctor, Dictionary`2 args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/ObjectCreator.cs:line 153
   at OpenRA.ObjectCreator.CreateObject[T](String className, Dictionary`2 args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/ObjectCreator.cs:line 113
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at OpenRA.Widgets.Widget.PostInit(WidgetArgs args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Widgets/Widget.cs:line 294
   at OpenRA.WidgetLoader.LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Widgets/WidgetLoader.cs:line 73
   at OpenRA.WidgetLoader.LoadWidget(WidgetArgs args, Widget parent, String w) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Widgets/WidgetLoader.cs:line 41
   at OpenRA.Widgets.Ui.OpenWindow(String id, WidgetArgs args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Widgets/Widget.cs:line 67
   at OpenRA.Mods.Common.Widgets.Logic.ConnectionLogic.ConnectionStateChanged(OrderManager om, String password, NetworkConnection connection) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Mods.Common/Widgets/Logic/ConnectionLogic.cs:line 39
   at OpenRA.Game.LogicTick() in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 645
   at OpenRA.Game.Loop() in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 814
   at OpenRA.Game.Run() in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 867
   at OpenRA.Game.InitializeAndRun(String[] args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 305
   at OpenRA.Launcher.Program.Main(String[] args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Launcher/Program.cs:line 32

**Comment by PunkPun** [comment]
weird, I thought I tested that

**Comment by PunkPun** [comment]
fixed

**Review by Mailaender — APPROVED** [review]

**Comment by ThomasChr** [comment]
Code looks good, tested on RA, works just fine:
image

**Review by ThomasChr — APPROVED** [review]
LGTM

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/28fe9b25b8161e3ffe3bc140d853be7cecaffb46...ca86f742643fbb3ec7aa278d444387f4387711c0)

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/commit/92d86494dc759958ffbdb920424ba938b8a4fd60
https://github.com/OpenRA/OpenRA/commit/e204080283c8abc99478c8180337f3def1e22ffd

**Pull request merged**